x-pack/plugins/data_enhanced/server/plugin.ts

```diff
@@ -51,6 +55,7 @@ export class EnhancedDataServerPlugin
     deps.data.search.registerSearchStrategy(
       ENHANCED_ES_SEARCH_STRATEGY,
       enhancedEsSearchStrategyProvider(
+        this.config$,
```

Comment:
Just FYI, config is available synchronously: `https://github.com/elastic/kibana/pull/88981`

Reply:
I'm going to make a small separate PR out of this change

Reply:
https://github.com/elastic/kibana/pull/90138